content/en/blog/_posts/2024-04-17-kubernetes-1.30.md

```diff
@@ -0,0 +1,218 @@
+safely without bumping the version of the object or breaking workflows.
+
+### Contextual logging ([SIG Instrumentation](https://github.com/kubernetes/community/tree/master/sig-instrumentation))
+
+Blurb for 3077 goes here.
+
+### Make Kubernetes aware of the LoadBalancer behaviour ([SIG Network](https://github.com/kubernetes/community/tree/master/sig-network))
+
+Blurb for 1860 goes here.
+
+## New alpha features
+
+### Speed up recursive SELinux label change ([SIG Storage](https://github.com/kubernetes/community/tree/master/sig-storage))
+
+This feature allows kubelet to apply SELinux label to volumes used in a pod in constant time using a mount option,  which speeds up pod startup on nodes with SELinux enabled. Without this feature, the container runtime had to recursively walk through the whole volumes and apply SELinux label individually to each file, which can be slow for volumes with large amount of files and directories.
+
+Kubernetes 1.27 graduated this feature as beta, but limited it to ReadWriteOncePod volumes. The corresponding feature gate is `SELinuxMountReadWriteOncePod`. It remains beta in 1.30.
+
+Kubernetes 1.30 extends support for SELinux mount option to all volumes as alpha, with a separate feature gate `SELinuxMount`. This feature gate introduces a behavioral change when multiple Pods with different SELinux labels share the same volume. See [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1710-selinux-relabeling#behavioral-changes) for details. We strongly encourage users that run  Kubernetes with SELinux enabled to test this feature and provide any feedback on the KEP.
+
+| Feature gate | Stage in v1.30 | Behavior change |
+|---|---|---| 
+| SELinuxMountReadWriteOncePod |  Beta | No |
+| SELinuxMount | Alpha | Yes | 
+
+Both feature gates `SELinuxMountReadWriteOncePod` and `SELinuxMount` must be enabled to test this feature on all volumes.
+
+This feature has no effect on Windows nodes or on Linux nodes without SELinux support.
+
+### Recursive Read-only (RRO) mounts ([SIG Node](https://github.com/kubernetes/community/tree/master/sig-node))
+
+Blurb for 3857 goes here--subject to Docs Freeze
+
+### Job success/completion policy ([SIG Apps](https://github.com/kubernetes/community/tree/master/sig-apps))
+
+Blurb for 3998 goes here--subject to Docs Freeze
+
+### Traffic distribution for services ([SIG Apps](https://github.com/kubernetes/community/tree/master/sig-apps))
+
+Blurb for 4444 goes here
+
+## Graduations, deprecations and removals for Kubernetes v1.30
+
+### Graduated to stable
+
+This lists all the features that graduated to stable (also known as _general availability_). For a
+full list of updates including new features and graduations from alpha to beta, see the [release
+notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.30.md).
+
+This release includes a total of XX enhancements promoted to Stable:
```

Comment:
```suggestion
This release includes a total of 16 enhancements promoted to stable:
```

Reply:
@kcmartin would love for you to double-check this number just in case I've miscounted!

Reply:
I see 17 on the project board. We'll do a final stats check for all the numbers we cite in the blog.